container-engine-lib/lib/backend_impls/docker/docker_kurtosis_backend/docker_kurtosis_backend.go

```diff
@@ -49,11 +50,7 @@ type DockerKurtosisBackend struct {
 	// This map is set exactly once, upon creation of the DockerKubernetesBackend, and never modified afterwards. Therefore, it doesn't need to be protected with a mutex (because the FreeIPProviders are themselves threadsafe)
 	enclaveFreeIpProviders map[enclave.EnclaveUUID]*free_ip_addr_tracker.FreeIpAddrTracker
 
-	// TODO Migrate this to an on-disk database, so that the API container can be shut down & restarted!
-	// Canonical store of the registrations being tracked by this *DockerKurtosisBackend instance
-	// NOTE: Unlike Kubernetes, Docker doesn't have a concrete object representing a service registration/IP address
-	//  allocation. We use this in-memory store to accomplish the same thing.
-	serviceRegistrations map[enclave.EnclaveUUID]map[service.ServiceUUID]*service.ServiceRegistration
+	serviceRegistrationRepository *service_registration.ServiceRegistrationRepository
 
 	// Control concurrent access to serviceRegistrations
 	serviceRegistrationMutex *sync.Mutex
```

Comment:
I think that we could remove this, but I didn't have enough time for testing it.